tools/test.py

```diff
@@ -100,6 +103,39 @@ def collect_results(result_part, size, tmpdir=None):
         return ordered_results
 
 
+def collect_results_gpu(result_part, size):
+    rank, world_size = get_dist_info()
+    # dump result part to tensor with pickle
+    part_tensor = torch.tensor(
+        bytearray(pickle.dumps(result_part)), dtype=torch.uint8, device='cuda')
+    # gather all result part tensor shape
+    shape_tensor = torch.tensor(part_tensor.shape, device='cuda')
+    shape_list = [shape_tensor.clone() for _ in range(world_size)]
+    dist.all_gather(shape_list, shape_tensor)
+    # padding result part tensor to max length
+    shape_max = torch.tensor(shape_list).max()
+    part_send = torch.zeros(shape_max, dtype=torch.uint8, device='cuda')
+    part_send[:shape_tensor[0]] = part_tensor
+    part_recv_list = [
+        part_tensor.new_zeros(shape_max) for _ in range(world_size)
+    ]
+    # gather all result part
+    dist.all_gather(part_recv_list, part_send)
+
+    if rank == 0:
+        part_list = [
+            pickle.loads(t[:shape_list[i][0]].cpu().numpy().tobytes())
+            for i, t in enumerate(part_recv_list)
+        ]
+        # sort the results
+        ordered_results = []
+        for res in zip(*part_list):
+            ordered_results.extend(list(res))
```

Comment:
You may confirm if this is necessary, or simply `mmcv.concat_list(part_list)` is enough.

Reply:
This is necessary, it concatenates the list of lists in a different manner from `mmcv.concat_list()`.